client/java/src/main/java/com/linecorp/centraldogma/client/Watcher.java

```diff
@@ -178,4 +183,33 @@ default void watch(Consumer<? super T> listener) {
         requireNonNull(listener, "listener");
         watch((revision, value) -> listener.accept(value));
     }
+
+    /**
+     * @return A new watcher, that reuses the supplied watcher and applies transformation.
+     * Note: the returned watcher is effectively filtering in a sense that,
+     * its listeners are *not* notified when a change has no effect on the transformed value.
+     */
+    default <U> Watcher<U> newChild(@Nonnull Function<T, U> transformer) {
+        return new TransformingWatcher<>(this, transformer);
+    }
+    /**
+     * @return A new child, which transformation is a JSON path query.
+     */
+    static Watcher<JsonNode> atJsonPath(@Nonnull Watcher<JsonNode> watcher, String jsonPath) {
+        return watcher.newChild(new Function<JsonNode, JsonNode>() {
+            @Override
+            public JsonNode apply(JsonNode node) {
+                if (node == null) {
+                    return MissingNode.getInstance();
+                } else {
+                    return node.at(jsonPath);
```

Comment:
I guess this is not a JSON path expression?

Reply:
Right, this is a JSON pointer. I got them mixed up.
It looks like JSON path is more powerful, I will look into it

Reply:
How about sticking to JSON pointer? (It's because JSON path implementation is relatively big and we don't want to introduce too many dependencies.)

Reply:
👍